File: test_save_score.js
const http = require('http');

function request(options, data = null) {
    return new Promise((resolve, reject) => {
        const req = http.request(options, res => {
            let body = '';
            res.on('data', chunk => body += chunk);
            res.on('end', () => resolve({ status: res.statusCode, body }));
        });
        req.on('error', reject);
        if (data) req.write(JSON.stringify(data));
        req.end();
    });
}

async function run() {
    try {
        const email = 'test' + Date.now() + '@example.com';
        let res = await request({
            hostname: 'localhost', port: 8080, path: '/api/auth/register', method: 'POST',
            headers: { 'Content-Type': 'application/json' }
        }, { firstName: 'Test', lastName: 'User', email, password: 'password' });
        
        if (res.status !== 200 && res.status !== 201) {
            console.log('Register failed', res);
            return;
        }
        
        const token = JSON.parse(res.body).token;
        console.log('Token:', token.substring(0, 20) + '...');
        
        const scoreRes = await request({
            hostname: 'localhost', port: 8080, path: '/api/aptitude/assessments/save-score', method: 'POST',
            headers: { 'Content-Type': 'application/json', 'Authorization': 'Bearer ' + token }
        }, { score: 10, totalQuestions: 15, accuracy: 66.67 });
        
        console.log('Score Response Status:', scoreRes.status);
        console.log('Score Response Body:', scoreRes.body);
    } catch (e) {
        console.error(e);
    }
}
run();

File: backend/TestSaveScore.java
package com.hirecraft.backend;

import java.net.URI;
import java.net.http.HttpClient;
import java.net.http.HttpRequest;
import java.net.http.HttpResponse;
import java.util.Map;
import com.fasterxml.jackson.databind.ObjectMapper;

public class TestSaveScore {
    public static void main(String[] args) throws Exception {
        HttpClient client = HttpClient.newHttpClient();
        ObjectMapper mapper = new ObjectMapper();
        
        // 1. Register
        Map<String, String> regBody = Map.of(
            "firstName", "T", "lastName", "U",
            "email", "t" + System.currentTimeMillis() + "@e.com",
            "password", "p"
        );
        HttpRequest regReq = HttpRequest.newBuilder()
            .uri(URI.create("http://localhost:8080/api/auth/register"))
            .header("Content-Type", "application/json")
            .POST(HttpRequest.BodyPublishers.ofString(mapper.writeValueAsString(regBody)))
            .build();
            
        HttpResponse<String> regResp = client.send(regReq, HttpResponse.BodyHandlers.ofString());
        if (regResp.statusCode() != 200 && regResp.statusCode() != 201) {
            System.out.println("Auth failed: " + regResp.statusCode() + " " + regResp.body());
            return;
        }
        
        String token = (String) mapper.readValue(regResp.body(), Map.class).get("token");
        
        // 2. Save score
        Map<String, Object> scoreBody = Map.of(
            "score", 10,
            "totalQuestions", 15,
            "accuracy", 66.67
        );
        HttpRequest scoreReq = HttpRequest.newBuilder()
            .uri(URI.create("http://localhost:8080/api/aptitude/assessments/save-score"))
            .header("Content-Type", "application/json")
            .header("Authorization", "Bearer " + token)
            .POST(HttpRequest.BodyPublishers.ofString(mapper.writeValueAsString(scoreBody)))
            .build();
            
        HttpResponse<String> scoreResp = client.send(scoreReq, HttpResponse.BodyHandlers.ofString());
        System.out.println("Score Response: " + scoreResp.statusCode() + " " + scoreResp.body());
    }
}
